go/lib/colibri/client/reservation.go

```diff
@@ -31,44 +29,59 @@ import (
 	"github.com/scionproto/scion/go/lib/sciond"
 	"github.com/scionproto/scion/go/lib/serrors"
 	"github.com/scionproto/scion/go/lib/snet"
+	"github.com/scionproto/scion/go/lib/spath"
 )
 
+// Reservation is an snet.Path like type that internally handles an e2e reservation.
+// Two use it, one must first create one through a call to NewReservation, and then
+// manage its lifetime with a call to Open and Close. Close must always be called
+// to free up the resources after the Reservation is no longer needed.
+// Internally, the Reservation has a periodic Runner handling the renewal of the reservation.
 type Reservation struct {
```

Comment:
What would happen if this reservation/path were used to:
- Contact a different destination machine in the same IA (in terms of `snet.path`, this path would be valid).
- Contact the same destination machine but a different port.

AFAIK both these should not work because the reservation is bound to a specific IA+machine+port. A flexible approach on `snet.Path` would be a function `canReachAddress(snet.UDPAddr)`, where each path can report for itself whether it can be used to reach a specific UDP address. Using this, an incompatible path can be detected before even attempting to send any data over the wire. The only current check for path-connection compatibility that I know of is [here](https://github.com/netsec-ethz/scion-apps/pull/187/files#diff-111f769b28036fcbf181cbd2fe04aa1076671858b5c1aa4697bf142d816cbbdcR80), where only the destination `IA` is checked.